### DISCUSSION TITLE
Add casting closures in eloquent models

### DISCUSSION THREAD

**Discussion by shivam-bmn** [laravel/framework#52494] [Ideas] [answered]
I was experimenting with some cool features in Laravel and noticed that one of my columns is storing multiple types of data, with the data type being stored in another column. I realized that I need to cast the data according to the type specified in that other column. I found that we can only use the `CastingAttributes` or `Castable` classes to create a new casting class.

I wanted to suggest that it might be useful if Laravel allowed us to use closures for casting attributes for simpler cases where we don’t need to do much beyond parsing the data type. This would provide a more streamlined approach for small, straightforward casting needs.

What do you think?

**Comment by henzeb** [comment]
InboundAttributes style?

**Reply by shivam-bmn** [reply]
Not sure what you mean, but I can give you my real example so you can get an idea about what actually I was trying to explain.

One of my table is containing the settings of the whole application. By whom, I can chabge the app appearance and other stuffs. The things is I was inserting the theme colors into a table. By which, I change the appearance, I have one column where I store the color value and a another column where I add type od that color like hex, rgb, rgba, hsl or other.

Thus was my concern that If laravel team can have any reference or implementation for this or something else they can provide. It's better

**Comment by shaedrich** [comment]
Additionally, there could be a `column` cast value, similar to when validation rules reference other columns, like `column:cast_type` where `cast_type` is the name of the column.

**Reply by henzeb** [reply]
Maybe you should show some code that reflects what you are looking for ...

**Reply by shaedrich** [reply, marked as answer]
```php
class Settings extends Model
{
    protected $casts = [
        'setting' => 'column:setting_type',
    ];
}
```

When the database looks like this:
| id |        setting        | setting_type |
| -- | --------------------- | ------------ |
|  1 |          1            |     int      |
|  2 |  2024-09-28 22:54:00  |     date     |
|  3 | lorem ipsum ahmet sun |    string    |

Then, the model will look like this:
```php
dump(Settings::find(1)->setting); // prints 1
dump(Settings::find(2)->setting); /* prints Carbon\Carbon @1727556978 {#1881 ▼
  #constructedObjectId: "0000000031bf18760000000007f7b216"
  #localMonthsOverflow: null
  #localYearsOverflow: null
  #localStrictModeEnabled: null
  #localHumanDiffOptions: null
  #localToStringFormat: null
  #localSerializer: null
  #localMacros: null
  #localGenericMacros: null
  #localFormatFunction: null
  #localTranslator: null
  #dumpProperties: array:3 [▶]
  #dumpLocale: null
  date: 2024-09-28 22:54:00.527354 UTC (+00:00)
} */
dump(Settings::find(3)->setting); // prints 'lorem ipsum ahmet sun'
```
